homeassistant/components/romy/vacuum.py

```diff
@@ -0,0 +1,212 @@
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.romy/.
+"""
+
+
+from collections.abc import Mapping
+from typing import Any
+
+from romy import RomyRobot
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, ICON, LOGGER
+from .coordinator import RomyVacuumCoordinator
+
+FAN_SPEED_NONE = "Default"
+FAN_SPEED_NORMAL = "Normal"
+FAN_SPEED_SILENT = "Silent"
+FAN_SPEED_INTENSIVE = "Intensive"
+FAN_SPEED_SUPER_SILENT = "Super_Silent"
+FAN_SPEED_HIGH = "High"
+FAN_SPEED_AUTO = "Auto"
+
+FAN_SPEEDS: list[str] = [
+    FAN_SPEED_NONE,
+    FAN_SPEED_NORMAL,
+    FAN_SPEED_SILENT,
+    FAN_SPEED_INTENSIVE,
+    FAN_SPEED_SUPER_SILENT,
+    FAN_SPEED_HIGH,
+    FAN_SPEED_AUTO,
+]
+
+# Commonly supported features
+SUPPORT_ROMY_ROBOT = (
+    VacuumEntityFeature.BATTERY
+    | VacuumEntityFeature.PAUSE
+    | VacuumEntityFeature.RETURN_HOME
+    | VacuumEntityFeature.SEND_COMMAND
+    | VacuumEntityFeature.STATUS
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.TURN_OFF
+    | VacuumEntityFeature.TURN_ON
+    | VacuumEntityFeature.FAN_SPEED
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ROMY vacuum cleaner."""
+
+    coordinator: RomyVacuumCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    romy: RomyRobot = coordinator.romy
+
+    device_info = {
+        "manufacturer": "ROMY",
+        "model": romy.model,
+        "sw_version": romy.firmware,
+        "identifiers": {"serial": romy.unique_id},
+    }
+
+    romy_vacuum_entity = RomyVacuumEntity(coordinator, romy, device_info)
+    entities = [romy_vacuum_entity]
+    async_add_entities(entities, True)
+
+
+class RomyVacuumEntity(CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity):
+    """Representation of a ROMY vacuum cleaner robot."""
+
+    def __init__(
+        self,
+        coordinator: RomyVacuumCoordinator,
+        romy: RomyRobot,
+        device_info: dict[str, Any],
+    ) -> None:
+        """Initialize the ROMY Robot."""
+        super().__init__(coordinator)
+        self.romy = romy
+        self._device_info = device_info
+        self._attr_unique_id = self.romy.unique_id
+        self._state_attrs: dict[str, Any] = {}
+
+        self._is_on = False
+        self._fan_speed = FAN_SPEEDS.index(FAN_SPEED_NONE)
+        self._fan_speed_update = False
+
+    @property
+    def supported_features(self) -> VacuumEntityFeature:
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_ROMY_ROBOT
```

Comment:
Please use the _attr shorthand for attributes which do not change
```suggestion
    self._attr_supported_features = SUPPORT_ROMY_ROBOT
```